### ISSUE TITLE
Feature Request: Replace part of selected word(s) with wildcards.

### ISSUE THREAD

**Issue by dsifford** [atom/find-and-replace#481]
Hi there. 

Not sure if this has been requested already (if it has, my apologies). I'd like the ability to replace parts of selected strings. Specifically, while leaving the inner part intact. 

Here is an example if that sounds confusing:

I'm debugging a program with variables shared between multiple files. The author of this code enumerated similar variables with numbers sandwiched in the center of the variable name (see below...)

```php
$variable_1_name
$variable_2_name
$variable_3_name

$different_1_name
$different_2_name
$different_3_name
```

What I'd like to do is rename this variable to something a little less ambiguous while keeping the numbering intact

I've searched using regex and found all occurrences of these variables, but (to my knowledge) I'm unable to use find and replace to either
1. Replace part of this selection (example 1, below)
2. Replace the outside parts of this selection (example 2, below)

Basically, what I'm visioning is using regex/wildcards for replacement. 

#### Example 1
Replace/delete just `_name` of each `$variable_x_name` without affecting `$different_x_name`. (replaced chunk in highlighed)

**Search Strategy:** `variable_\d_name`
**Replace Strategy:** `\w{8}_\d` 
**Result:** 
```php
$variable_1
$variable_2
$variable_3
```

#### Example 2
Replace `variable` _ x _ `name`, while leaving the `_x_` in place. 

**Search Strategy:** `variable_\d_name`
**Replace Strategy:** `clearer_%*%_name`
**Result:**
```php
$clearer_1_name
$clearer_2_name
$clearer_3_name
```

If this is already possible, could you direct me how it's done? If not, perhaps consider adding? 

Thanks for all your work!

**Comment by benogle** [comment]
You can use `$#` syntax in replacements to use bits from the found text, where `$1` would be the first regex group. See [the js regex docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter) for more information. Basically: 

```
Find Pattern: variable_(\d+)_name
Replace Pattern: variable_$1
```

**Comment by dsifford** [comment]
Excellent! Thanks for the record-speed reply!

Cheers!

**Comment by benogle** [comment]
:+1:
